addons/web/static/src/core/install_prompt/install_prompt_service.js

```diff
@@ -0,0 +1,79 @@
+/** @odoo-module **/
+
+import { reactive } from "@odoo/owl";
+import { browser } from "@web/core/browser/browser";
+import { isDisplayStandalone, isIOS, isMacOS } from "@web/core/browser/feature_detection";
+import { registry } from "@web/core/registry";
+import { InstallPrompt } from "./install_prompt";
+
+const serviceRegistry = registry.category("services");
+
+const installPromptService = {
+    dependencies: ["dialog"],
+    start(env, { dialog }) {
+        let nativePrompt;
+
+        const state = reactive({
+            canPromptToInstall: false,
+            decline,
+            show,
+        });
+
+        // The PWA can only be installed if the app is not already launched
+        // The installation is not possible from the tests by default
+        // Finally, PWA are supported on desktop Safari since version 17
+        const canBeInstalled =
+            !isDisplayStandalone() &&
+            browser.location.pathname !== "/web/tests" &&
```

Comment:
This was to avoid having to mess with mail tests with the notification area, which would fail on devices having the PWA installed, or if the browser don't support them (since assertions would fail about the number of notifications)

Reply:
Not really sure about this... I'll invoke the high authority of sir @aab-odoo :grin: for a second opinion